keps/sig-release/20190118-creating-packages.md

```diff
@@ -0,0 +1,417 @@
+
+<!--
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+-->
+
+<!--
+**ACTION REQUIRED:** There must be an issue in
+[kubernetes/enhancements](https://github.com/kubernetes/enhancements/issues)
+referencing this KEP and targeting a release milestone **before [Enhancement
+Freeze](https://github.com/kubernetes/sig-release/tree/master/releases) of the
+targeted release**.
+-->
+
+## Release Signoff Checklist
+
+Check these off as they are completed for the Release Team to track. These checklist items _must_ be updated for the enhancement to be released.
+
+- [ ] k/enhancements issue in release milestone and linked to KEP (insert link here)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+
+## Summary
+
+There are currently different ways to build packages (`deb`s and
+`rpm`s) for the kubernetes components. All of which have some shortcomings and
+it is hard to understand for users which one is the correct one.
+The goal is to streamline all the package building mechanism, create one way on
+how to generate packages, and retire all the other options.
+
+## Motivation
+
+As mentioned in the summary, there are multiple different ways on how to create
+packages for kubernetes:
+
+- [`k/release/debian`][k-release-deb] & [`k/release/rpm`][k-release-rpm]
+  - *Description*:  
+    These tools are part of the official release process. After a new
+    kubernetes version has been cut and released, but before the new release
+    was announced, these tools will pull the build binaries from a bucket and
+    wrap them in a `deb`/`rpm`.
+  - *Usage*:
+    ```
+    k/release # ./debian/jenkins.sh
+    k/release # cd ./rpm/ && ./docker-build.sh
+    ```
+  - *Cons*<a id="k-release-cons"></a>:
+    - It uses a different (no) branching strategy then the artefacts build from `k/k` it is packaging
+    - Pulls pre-existing artefacts from a GCS bucket
+    - Cannot build just one package, builds all the packages for all architectures (on linux)
+    - Needs manual change of the source code when bumping versions
+  - *Pros*:
+    - builds for multiple platforms (`linux/aarch64`, `linux/armhfp`, `linux/ppc64le`, `linux/s390x`, `linux/x86_64`)
+    - also handles the upload of the packages (at least for `deb`s)
+      - might also be considered a [con](#k-release-cons)?
+  - *Parts to reuse*:
+    - multi-arch support
+- [`k/k/debs`][k-k-deb] & [`k/k/rpms`][k-k-rpm]
+  - *Description*:  
+    This is building the packages directly with bazel. Those packages are not
+    used as official releases -- usecase is not clear.
```

Comment:
currently this is used in some CI. original usecase was potentially superseding the others.